### PR TITLE
[FLINK-26704][table] Remove string expressions

### PR DESCRIPTION
**Pull request by slinkydeveloper**
This PR removes all the code related to Java String expressions DSL.

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/table/validation/CalcValidationTest.scala [hunk 1]
```diff
@@ -108,7 +108,7 @@ class CalcValidationTest extends TableTestBase {
   def testDropColumnsWithValueLiteral(): Unit = {
     val util = streamTestUtil()
     val tab = util.addTableSource[(Int, Long, String)]("Table3",'a, 'b, 'c)
-    tab.dropColumns("'a'")
+    tab.dropColumns("a")
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/table/validation/CalcValidationTest.scala:111]: why is this still a string and not a ref?

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/CompositeTypeTestBase.scala [hunk 2]
```diff
@@ -47,16 +47,16 @@ class CompositeTypeTestBase extends ExpressionTestBase {
 
   override def typeInfo: RowTypeInfo = {
     new RowTypeInfo(
-      /* 0 */  createTypeInformation[MyCaseClass],
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/CompositeTypeTestBase.scala:50]: undo changes in this file for now

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala [hunk 3]
```diff
@@ -214,23 +211,23 @@ abstract class ExpressionTestBase {
   }
 
   def testExpectedTableApiException(
-      expr: Expression,
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala:217]: double space indention

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala [hunk 4]
```diff
@@ -179,9 +175,9 @@ abstract class ExpressionTestBase {
   }
 
   def testAllApis(
-      expr: Expression,
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala:182]: double space indention

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala [hunk 5]
```diff
@@ -304,23 +296,23 @@ abstract class ExpressionTestBase {
   }
 
   private def addTableApiTestExpr(
-      tableApiExpr: Expression,
-      expected: String,
-      exprsContainer: mutable.ArrayBuffer[_],
-      exceptionClass: Class[_ <: Throwable] = null): Unit = {
+       tableApiExpr: Expression,
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala:299]: double space indention

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala [hunk 6]
```diff
@@ -304,23 +296,23 @@ abstract class ExpressionTestBase {
...
     addTestExpr(relNode, expected, tableApiExpr.asSummaryString(), null, exprsContainer)
   }
 
   private def addTestExpr(
-      relNode: RelNode,
-      expected: String,
-      summaryString: String,
-      exceptionClass: Class[_ <: Throwable],
-      exprs: mutable.ArrayBuffer[_]): Unit = {
+       relNode: RelNode,
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ExpressionTestBase.scala:311]: double space indention

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/MapTypeTestBase.scala [hunk 7]
```diff
@@ -59,16 +59,16 @@ abstract class MapTypeTestBase extends ExpressionTestBase {
 
   override def typeInfo: RowTypeInfo = {
     new RowTypeInfo(
-      /* 0 */  new MapTypeInfo(BasicTypeInfo.STRING_TYPE_INFO, BasicTypeInfo.STRING_TYPE_INFO),
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/MapTypeTestBase.scala:62]: undo changes in this file for now

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/RowTypeTestBase.scala [hunk 8]
```diff
@@ -57,9 +57,9 @@ abstract class RowTypeTestBase extends ExpressionTestBase {
       /* 2 */ Types.ROW(Types.INT, Types.STRING, Types.BOOLEAN),
       /* 3 */ Types.ROW(Types.INT, Types.ROW(Types.INT, Types.STRING, Types.BOOLEAN)),
       /* 4 */ Types.ROW(
-                Types.LOCAL_DATE,
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/RowTypeTestBase.scala:60]: undo changes in this file for now

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ScalarOperatorsTestBase.scala [hunk 9]
```diff
@@ -59,32 +59,32 @@ abstract class ScalarOperatorsTestBase extends ExpressionTestBase {
...
+      ),
+      DataTypes.FIELD("f14", DataTypes.STRING()),
+      DataTypes.FIELD("f15", DataTypes.DATE()),
+      DataTypes.FIELD("f16", DataTypes.DECIMAL(19, 8)),
+      DataTypes.FIELD("f17", DataTypes.DECIMAL(19, 1)),
+      DataTypes.FIELD("f18", DataTypes.BINARY(200)),
+      DataTypes.FIELD("f19", DataTypes.VARBINARY(200)),
+      DataTypes.FIELD("f20", DataTypes.VARBINARY(200)),
+      DataTypes.FIELD("f21", DataTypes.TIME()),
+      DataTypes.FIELD("f22", DataTypes.TIMESTAMP())
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ScalarOperatorsTestBase.scala:87]: undo changes in this file for now

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ScalarTypesTestBase.scala [hunk 10]
```diff
@@ -96,16 +96,16 @@ abstract class ScalarTypesTestBase extends ExpressionTestBase {
 
   override def typeInfo: RowTypeInfo = {
     new RowTypeInfo(
-      /* 0 */  Types.STRING,
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/ScalarTypesTestBase.scala:99]: undo changes in this file for now

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/userDefinedScalarFunctions.scala [hunk 11]
```diff
@@ -203,8 +203,8 @@ object Func24 extends ScalarFunction {
 
 
 /**
-  * A scalar function that always returns TRUE if opened correctly.
-  */
+ * A scalar function that always returns TRUE if opened correctly.
+ */
 @SerialVersionUID(1L)
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/utils/userDefinedScalarFunctions.scala:208]: undo changes in this file for now

File: flink-python/pyflink/table/tests/test_pandas_udaf.py [hunk 12]
```diff
@@ -107,10 +107,10 @@ def test_group_aggregate_with_aux_group(self):
...
                                                      func_type="pandas"))
         self.t_env.create_temporary_system_function("mean_udaf", mean_udaf)
-        t.group_by("a") \
-            .select("a, a + 1 as b, a + 2 as c") \
-            .group_by("a, b") \
-            .select("a, b, mean_udaf(b), max_add(b, c, 1)") \
+        t.group_by(t.a) \
+            .select(t.a,  (t.a + 1).alias("b"), (t.a + 2).alias("c")) \
+            .group_by(t.a, t.b) \
+            .select(t.a, t.b, mean_udaf(t.b), expr.call("max_add", t.b, t.c, 1)) \
```
Comment (twalthr) [inline flink-python/pyflink/table/tests/test_pandas_udaf.py:113]: can't we import `expr` as we did in `flink-end-to-end-tests/flink-python-test/python/python_job.py` to call `call` and `col` directly?

**Review by twalthr — CHANGES_REQUESTED** [review]
Nice work @slinkydeveloper. This makes the code base so much cleaner.

**Review by twalthr — APPROVED** [review]
LGTM

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/table/validation/CalcValidationTest.scala (hunk 1 above)
Comment (slinkydeveloper) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/table/validation/CalcValidationTest.scala:111]: This is actually correct, as it's testing that the string literal doesn't work

**Pull request closed without merging**